cvmfs/sync_item.cc

```diff
@@ -185,4 +192,81 @@ std::string SyncItem::GetScratchPath() const {
   return union_engine_->scratch_path() + relative_path;
 }
 
+
+std::string SyncItem::GetGraftPath() const {
+  return union_engine_->scratch_path() + "/" + ((relative_parent_path_.empty()) ?
+      ".cvmfsgraft-" + filename_ :
+      relative_parent_path_ + (filename_.empty() ? "" : ("/.cvmfsgraft-" + filename_)));
+}
+
+void SyncItem::CheckGraft() {
+  valid_graft_ = false;
+  bool found_checksum = false;
+  std::string checksum_type;
+  std::string checksum_value;
+  std::string graftfile = GetGraftPath();
+  LogCvmfs(kLogFsTraversal, kLogDebug, "Checking potential graft path %s.", graftfile.c_str());
+  FILE *fp = fopen(graftfile.c_str(), "r");
+  if (fp == NULL) {
+    if (errno != ENOENT) {
+      LogCvmfs(kLogFsTraversal, kLogWarning, "Unable to open graft file (%s): %s (errno=%d)",
+               graftfile.c_str(), strerror(errno), errno);
+    }
+    return;
+  }
+  graft_marker_present_ = true;
+  size_t len = 0;
+  ssize_t read;
```

Comment:
We have a few utility functions in util.cc that should be more handy here.  In particular, `GetLineFile()` and `SplitString()` should make this paragraph shorter.